test/integration/targets/jmespath/tasks/main.yml

```diff
@@ -0,0 +1,4 @@
+- name: Install JMESPath
```

Comment:
Generally for external dependencies tests use a virtual environment rather than modifying the test nodes.
You could add some tasks to set that up like https://github.com/ansible/ansible/blob/924352a051b797b0e8eabec1966b9d985f2ca9b8/test/integration/targets/ecs_cluster/tasks/main.yml, or you could add a runme.sh script like https://github.com/ansible/ansible/blob/0c74ee4352fc5702de25d8bab8a592e0e044f0bc/test/integration/targets/filters/runme.sh that then calls your test playbook.